uwblueprint/extend-a-family
# implement course units api

## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/7e51fde5f22d448b8226f2cfa4dd93d5?v=b2dc0dba06574651a8b6063bdd0d74e0&p=6985a67379d34a188643cd555a4945d6&pm=s)



## Implementation description
* Implemented course units api for the following functions:
* GET: Gets all units
* POST: Creates a unit
* PUT: updates a unit by id
* DELETE: deletes a unit by id




## Steps to test
You should probably run step 2 everytime after an post, put and delete request. or you can just check MongoDB instead to ensure everything is correct,

1. Obtain a bearer token for an administrator account
2. Run the get endpoint at /courseUnits -> take note of how many units there are as well as the largest displayIndex value
4. Run post endpoint at /courseUnits with title as the only body parameter
5. Take any id that is valid and update it by running the put endpoint at /courseUnits/:id and give it a different title
6. Take the id of a courseUnit with the largest displayIndex and delete it by running the put endpoint at /courseUnits/:id, make sure that the other course units display index DON'T CHANGE
7. Do the same but with a courseUnit with a displayIndex that isn't the largest and delete, for every course unit with a display index larger than it, they should all be decreased by 1.



## What should reviewers focus on?
* Check my mongoose queries in deleteCourse and createCourse. Is this the best way of doing what is mentioned in the ticket?


## Checklist
